net/ip.cpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2023 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <cmath>
+#include <cstdint>
+#include <cstring>
+#include <span>
+
+#include <fmt/core.h>
+
+#include "ip.h"
```

Comment:
`net/ip.h`. We use the full path everywhere else, and the header order (that will be enforced by clang-tidy soon(tm)) is
1. main header (the header being tested, or the header the .cpp is for)
2. our other headers
3. 3rd-party
4. stdlib